src/connector/src/parser/protobuf/parser.rs

```diff
@@ -525,19 +525,24 @@ fn protobuf_type_mapping(
     Ok(t)
 }
 
+/// Reference: <https://docs.confluent.io/platform/current/schema-registry/fundamentals/serdes-develop/index.html#wire-format>
+/// Wire format for Confluent pb header is:
+/// | 0          | 1-4        | 5-x             | x+1-end
+/// | magic-byte | schema-id  | message-indexes | protobuf-payload
 pub(crate) fn resolve_pb_header(payload: &[u8]) -> Result<&[u8]> {
     // there's a message index array at the front of payload
     // if it is the first message in proto def, the array is just and `0`
     // TODO: support parsing more complex index array
     let (_, remained) = extract_schema_id(payload)?;
+    // The message indexes are encoded as int using variable-length zig-zag encoding,
+    // prefixed by the length of the array.
+    // Note that if the first byte is 0, it is equivalent to (1, 0) as an optimization.
     match remained.first() {
         Some(0) => Ok(&remained[1..]),
-        Some(i) => {
-            Err(RwError::from(ProtocolError(format!("The payload message must be the first message in protobuf schema def, but the message index is {}", i))))
-        }
-        None => {
-            Err(RwError::from(ProtocolError("The proto payload is empty".to_owned())))
-        }
+        Some(i) => Ok(&remained[(*i as usize)..]),
```

Comment:
This is skipping `i` **bytes** rather than skipping `i` **variable-length zig-zag encoded integers**. For `i` within `1..=63`, the encoding outputs a single byte so this happens to work.

Reply:
Got it, so the correct implementation is
```
        Some(i) => Ok(&remained[(1 + *i / 2) as usize..]),
```